src/messages/embed.ts

```diff
@@ -0,0 +1,283 @@
+import {
+	APIEmbed,
+	APIEmbedAuthor,
+	APIEmbedField,
+	APIEmbedFooter,
+	APIEmbedImage,
+	APIEmbedProvider,
+	APIEmbedThumbnail,
+	APIEmbedVideo,
+	EmbedType,
+} from 'discord-api-types/v8';
+
+/**
+ * Represents an embed in a message (image/video preview, rich embed, etc.)
+ */
+export class Embed implements APIEmbed {
+	/**
+	 * Embed Fields.
+	 */
+	public fields: APIEmbedField[] = [];
+
+	/**
+	 * The type of embed.
+	 * @since 0.0.1
+	 */
+	public type?: EmbedType;
+
+	/**
+	 * The embed title.
+	 */
+	public title?: string;
+
+	/**
+	 * The embed description.
+	 */
+	public description?: string;
+
+	/**
+	 * The embed url.
+	 */
+	public url?: string;
+
+	/**
+	 * The embed bar color.
+	 */
+	public color?: number;
+
+	/**
+	 * The timestamp of the embed.
+	 */
+	public timestamp?: string;
+
+	/**
+	 * The embed thumbnail data.
+	 */
+	public thumbnail?: APIEmbedThumbnail;
+
+	/**
+	 * The embed image data.
+	 * @since 0.0.1
+	 */
+	public image?: APIEmbedImage;
+
+	/**
+	 * Received video data.
+	 * @since 0.0.1
+	 */
+	public video?: APIEmbedVideo;
+
+	/**
+	 * The embed author data.
+	 * @since 0.0.1
+	 */
+	public author?: APIEmbedAuthor;
+
+	/**
+	 * Received data about the embed provider.
+	 * @since 0.0.1
+	 */
+	public provider?: APIEmbedProvider;
+
+	public footer?: APIEmbedFooter;
+
+	public constructor(data: APIEmbed = {}) {
+		this.type = data.type;
+		this.title = data.title;
+		this.description = data.description;
+		this.url = data.url;
+		this.color = data.color;
+		this.thumbnail = data.thumbnail;
+		this.image = data.image;
+		this.video = data.video;
+		this.author = data.author;
+		this.provider = data.provider;
+		this.footer = data.footer;
+
+		if (data.timestamp) this.timestamp = new Date(data.timestamp).toISOString();
+		if (data.fields) this.fields = data.fields;
+	}
```

Comment:
No other builder so far parses data you pass in... @discordjs/the-big-3 thoughts?

Reply:
Resolved, see https://github.com/discordjs/builders/pull/11#issuecomment-883655746